catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -845,15 +591,449 @@ function DropzoneMessage({ error, warn, disabled }: DropzoneMessageProps) {
   )
 }
 
-interface FilesInputProps {
-  input: {
-    value: FilesState
-    onChange: (value: FilesState) => void
-  }
-  className?: string
-  disabled?: boolean
-  errors?: Record<string, React.ReactNode>
-  meta: {
+const useRootStyles = M.makeStyles({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+})
+
+function Root({ className, ...props }: React.PropsWithChildren<{ className?: string }>) {
+  const classes = useRootStyles()
+  return <div className={cx(classes.root, className)} {...props} />
+}
+
+const useHeaderStyles = M.makeStyles({
+  root: {
+    display: 'flex',
+    height: 24,
+  },
+})
+
+function Header(props: React.PropsWithChildren<{}>) {
+  const classes = useHeaderStyles()
+  return <div className={classes.root} {...props} />
+}
+
+const useHeaderTitleStyles = M.makeStyles((t) => ({
+  root: {
+    ...t.typography.body1,
+    alignItems: 'center',
+    display: 'flex',
+    overflow: 'hidden',
+    whiteSpace: 'nowrap',
+  },
+  regular: {},
+  disabled: {
+    color: t.palette.text.secondary,
+  },
+  error: {
+    color: t.palette.error.main,
+  },
+  warn: {
+    color: t.palette.warning.dark,
+  },
+}))
+
+type HeaderTitleState = 'disabled' | 'error' | 'warn' | 'regular'
+
+function HeaderTitle({
+  state = 'regular',
+  ...props
+}: React.PropsWithChildren<{ state?: HeaderTitleState }>) {
+  const classes = useHeaderTitleStyles()
+  return <div className={cx(classes.root, classes[state])} {...props} />
+}
+
+const PROGRESS_EMPTY = { total: 0, loaded: 0, percent: 0 }
+
+const useLockStyles = M.makeStyles((t) => ({
+  root: {
+    alignItems: 'center',
+    background: 'rgba(255,255,255,0.9)',
+    border: `1px solid ${t.palette.action.disabled}`,
+    borderRadius: t.shape.borderRadius,
+    bottom: 0,
+    cursor: 'not-allowed',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    left: 0,
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  progressContainer: {
+    display: 'flex',
+    position: 'relative',
+  },
+  progressPercent: {
+    ...t.typography.h5,
+    alignItems: 'center',
+    bottom: 0,
+    display: 'flex',
+    justifyContent: 'center',
+    left: 0,
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  progressSize: {
+    ...t.typography.body2,
+    color: t.palette.text.secondary,
+    marginTop: t.spacing(1),
+  },
+}))
+
+function Lock({
+  progress,
+}: {
+  progress?: {
+    total: number
+    loaded: number
+    percent: number
+  }
+}) {
+  const classes = useLockStyles()
+  return (
+    <div className={classes.root}>
+      {!!progress && (
```

Comment:
Early exit will make code slightly clearer:

```tsx
  const classes = useLockStyles()

  if (!progress) return <div className={classes.root} />

  return (
    <div className={classes.root}>
      <div className={classes.progressContainer}>
        …
      </div>
    </div>
  )
```

Reply:
i dont think so bc in this particular case it adds repetition, while the condition just hides some nested elements, which seems idiomatic to me